### DISCUSSION TITLE
helmfile.yaml.gotmpl isn't detected by helmfile manager

### DISCUSSION THREAD

**Discussion by z0rc** [renovatebot/renovate#23166] [Suggest an Idea]
### Tell us more.

Describe the problem

Helmfile is moving from helmfile.yaml to helmfile.yaml.gotmpl. Reasoning why described at [helmfile/helmfile@main/docs/proposals/towards-1.0.md#force-gotmpl-file-extension-for-helmfileyaml-in-case-you-want-helmfile-to-render-it-as-a-go-template-before-yaml-parsing](https://github.com/helmfile/helmfile/blob/main/docs/proposals/towards-1.0.md#force-gotmpl-file-extension-for-helmfileyaml-in-case-you-want-helmfile-to-render-it-as-a-go-template-before-yaml-parsing). Right now helmfile v0 accepts both .yaml and .yaml.gotmpl files ([helmfile/helmfile@main/pkg/app/app.go#L1213](https://github.com/helmfile/helmfile/blob/main/pkg/app/app.go#L1213)), this will change to just yaml.gotmpl with helmfile v1.

Minimal reproduction: [z0rc/renovate-helmfile-yaml-gotmpl](https://github.com/z0rc/renovate-helmfile-yaml-gotmpl)

**Comment by rarkins** [comment]
Users might be able to change `fileMatch` already to capture the new name.

The challenge is what to do with the Go templating itself.

Renovate today uses an off-the-shelf YAML parser, with some basic attempts to strip out templating first: https://github.com/renovatebot/renovate/blob/2e497c8acb43aba5c7cafa36d5235c445839244b/lib/modules/manager/helmfile/extract.ts#L21-L26

Questions:
- Is this stripping sufficient for the majority of cases? In which case we can extend the default `fileMatch` immediately and don't need to do anything else
- Do we need more stripping, or do we need _compiling_? If compiling, that's a whole new level of complexity, for example we may need to bundle a Go binary, perhaps allow inputs to the compilation, etc.

**Reply by z0rc** [reply]
Current state of things:
- helmfile v0 treats `helmfile.yaml` and `helmfile.yaml.gotmpl` equally as go template that rendered into yaml
- helmfile v1 won't allow go template constructs in `helmfile.yaml` and considers this as an error, `helmfile.yaml.gotmpl` will be treated the same way as in helmfile v0
- renovate handles `helmfile.yaml` as helmfile v0, go template constructs are cleaned up from file in hope that result will be a parseable yaml

I believe currently renovate can extend default `fileMatch` for helmfile to allow handling `helmfile.yaml.gotmpl` the same way as `helmfile.yaml`. After some time of helmfile v1 release you might want to keep go template cleanup logic only for `helmfile.yaml.gotmpl` and expect `helmfile.yaml` to be regular yaml file.

**Comment by rarkins** [comment]
Feature request: #23176

**Comment by pathob** [comment]
Duplicate of #23164 and implemented with #26063. Can be closed.
